fish-rust/src/ast.rs

```diff
@@ -0,0 +1,5206 @@
+            self.as_node().as_argument().unwrap()
+        }
+        fn as_begin_header(&self) -> &BeginHeader {
+            self.as_node().as_begin_header().unwrap()
+        }
+        fn as_block_statement(&self) -> &BlockStatement {
+            self.as_node().as_block_statement().unwrap()
+        }
+        fn as_decorated_statement(&self) -> &DecoratedStatement {
+            self.as_node().as_decorated_statement().unwrap()
+        }
+        fn as_for_header(&self) -> &ForHeader {
+            self.as_node().as_for_header().unwrap()
+        }
+        fn as_freestanding_argument_list(&self) -> &FreestandingArgumentList {
+            self.as_node().as_freestanding_argument_list().unwrap()
+        }
+        fn as_function_header(&self) -> &FunctionHeader {
+            self.as_node().as_function_header().unwrap()
+        }
+        fn as_if_statement(&self) -> &IfStatement {
+            self.as_node().as_if_statement().unwrap()
+        }
+        fn as_job_conjunction_continuation(&self) -> &JobConjunctionContinuation {
+            self.as_node().as_job_conjunction_continuation().unwrap()
+        }
+        fn as_job_continuation(&self) -> &JobContinuation {
+            self.as_node().as_job_continuation().unwrap()
+        }
+        fn as_job_list(&self) -> &JobList {
+            self.as_node().as_job_list().unwrap()
+        }
+        fn as_not_statement(&self) -> &NotStatement {
+            self.as_node().as_not_statement().unwrap()
+        }
+        fn as_redirection(&self) -> &Redirection {
+            self.as_node().as_redirection().unwrap()
+        }
+        fn as_statement(&self) -> &Statement {
+            self.as_node().as_statement().unwrap()
+        }
+        fn as_switch_statement(&self) -> &SwitchStatement {
+            self.as_node().as_switch_statement().unwrap()
+        }
+        fn as_while_header(&self) -> &WhileHeader {
+            self.as_node().as_while_header().unwrap()
+        }
+}
+
+impl StatementVariant {
+    fn ptr(&self) -> Box<NodeFFI<'_>> {
+        match self {
+            StatementVariant::None => panic!(),
+            StatementVariant::NotStatement(node) => node.ptr(),
+            StatementVariant::BlockStatement(node) => node.ptr(),
+            StatementVariant::IfStatement(node) => node.ptr(),
+            StatementVariant::SwitchStatement(node) => node.ptr(),
+            StatementVariant::DecoratedStatement(node) => node.ptr(),
+        }
+    }
+}
+impl BlockStatementHeaderVariant {
+    fn ptr(&self) -> Box<NodeFFI<'_>> {
+        match self {
+            BlockStatementHeaderVariant::None => panic!(),
+            BlockStatementHeaderVariant::ForHeader(node) => node.ptr(),
+            BlockStatementHeaderVariant::WhileHeader(node) => node.ptr(),
+            BlockStatementHeaderVariant::FunctionHeader(node) => node.ptr(),
+            BlockStatementHeaderVariant::BeginHeader(node) => node.ptr(),
+        }
+    }
+}
+
+impl AndorJobList {
```

Comment:
We might as well correct the capitalization and mkae this `AndOrXxx` everywhere.